### PR TITLE
Model run on pr action 

### PR DESCRIPTION
**Pull request by ZacharyWills**
#233 

Fixes Workflow for running example model based on PR code and not the existing.

Note the new contributors will need maintainer help to run initial PR workflow after review:
https://github.blog/changelog/2021-04-22-github-actions-maintainers-must-approve-first-time-contributer-workflow-runs/

## Additions
Additional Dockerfiles for running nGen on various CentOs configurations, and updates the existing one run by the workflow to copy in the PR code and not reclone from the main

-

## Removals

-

## Changes
Workflow file now expects to run the PR code
-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ X] PR has an informative and human-readable title
- [X ] Changes are limited to a single goal (no scope creep)
- [X ] Code can be automatically merged (no conflicts)
- [X ] Code follows project standards (link if applicable)
- [X ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [X ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1. Should make sure that subsequent PRs pull the PR branch and run it

### Target Environment support

- [X] Linux
- [X] macOS

### CODE REVIEW

File: docker/CENTOS_4.8.5_NGEN_RUN.dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+RUN git submodule update --init --recursive -- test/googletest
+
+RUN curl -L -O https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2
+
+RUN tar -xjf boost_1_72_0.tar.bz2
+
+ENV BOOST_ROOT="boost_1_72_0"
+
+WORKDIR /ngen
```
Comment (hellkite500) [inline docker/CENTOS_4.8.5_NGEN_RUN.dockerfile:28]: Redundant WORKDIR

File: docker/CENTOS_4.8.5_NGEN_RUN.dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+RUN cmake -B /ngen -S .
+
+RUN cmake --build /ngen --target ngen
+
+WORKDIR /ngen/
```
Comment (hellkite500) [inline docker/CENTOS_4.8.5_NGEN_RUN.dockerfile:34]: And again.  WORKDIR hasn't changed

File: docker/CENTOS_latest_NGEN_RUN.dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+FROM centos:8 as builder
```
Comment (hellkite500) [inline docker/CENTOS_latest_NGEN_RUN.dockerfile:1]: This dockerfile is named `CENTOS_latest` but builds from centos8 whereas `CENTOS_NGEN_RUN` actually pulls `centos:latest`

**Review by hellkite500 — COMMENTED** [review]
Where there no changes needed in the action workflow file?

File: docker/CENTOS_latest_NGEN_RUN.dockerfile (hunk 3 above)
Comment (ZacharyWills) [reply docker/CENTOS_latest_NGEN_RUN.dockerfile:1]: Weirdly enough the :latest and :8 both pull the latest Centos8, but the :8 won't upgrade when the inevitable Centos 9 is out. I maybe need to label the 8_Latest better.

File: docker/CENTOS_4.8.5_NGEN_RUN.dockerfile (hunk 1 above)
Comment (ZacharyWills) [reply docker/CENTOS_4.8.5_NGEN_RUN.dockerfile]: fixed

File: docker/CENTOS_4.8.5_NGEN_RUN.dockerfile (hunk 2 above)
Comment (ZacharyWills) [reply docker/CENTOS_4.8.5_NGEN_RUN.dockerfile]: fixed

**Review by hellkite500 — APPROVED** [review]

**Pull request merged**